contracts/Nexus.sol

```diff
@@ -140,6 +142,14 @@ contract Nexus is INexus, EIP712, BaseAccount, ExecutionHelper, ModuleManager, U
             if (execType == EXECTYPE_DEFAULT) returnData = _executeBatch(executions);
             else if (execType == EXECTYPE_TRY) returnData = _tryExecuteBatch(executions);
             else revert UnsupportedExecType(execType);
+        } else if (callType == CALLTYPE_DELEGATECALL) {
+            // destructure executionCallData according to single exec
+            address delegate = address(uint160(bytes20(executionCalldata[0:20])));
+            bytes calldata callData = executionCalldata[20:];
+            // check if execType is revert or try
+            if (execType == EXECTYPE_DEFAULT) _executeDelegatecall(delegate, callData);
+            else if (execType == EXECTYPE_TRY) _tryExecuteDelegatecall(delegate, callData);
+            else revert UnsupportedExecType(execType);
```

Comment:
Can we just use `_handleDelegateCallExecution()` here?

Reply:
I see.. ok

Reply:
in executeFromExecutor it's not used anywhere in this sense. so i kept this way

Reply:
maybe can refactor the whole method then? :) since we already have those _handle____ methods, why not use them anywhere?

Reply:
can even move all of those methods into library, and do like`executionCalldata._handleSingleExecution(execType)`

Reply:
yes we do, can do. one required return data so I have to check again how to best refactor

Reply:
library ExecLib has all pure methods. 
in order to do this: executionCalldata._handleSingleExecution(execType)
I'd have to make a new base contract that is able to use internal methods from ExecutionHelper.sol